datafusion/sqllogictest/test_files/information_schema.slt

```diff
@@ -115,14 +115,14 @@ datafusion information_schema schemata VIEW
 datafusion information_schema tables VIEW
 datafusion information_schema views VIEW
 
-query TTTT rowsort
-SELECT * from information_schema.tables WHERE datafusion.information_schema.tables.table_schema='information_schema';
-----
-datafusion information_schema columns VIEW
-datafusion information_schema df_settings VIEW
-datafusion information_schema schemata VIEW
-datafusion information_schema tables VIEW
-datafusion information_schema views VIEW
+#query TTTT rowsort
+#SELECT * from information_schema.tables WHERE datafusion.information_schema.tables.table_schema='information_schema';
+#----
+#datafusion information_schema columns VIEW
+#datafusion information_schema df_settings VIEW
+#datafusion information_schema schemata VIEW
+#datafusion information_schema tables VIEW
+#datafusion information_schema views VIEW
```

Comment:
Was there a reason this is being commented?

Reply:
Thanks @Jefffrey 
ideally I think this query shouldn't work. While debugging I found the next 
```
Error: SchemaError(FieldNotFound { field: Column { relation: Some(Full { catalog: "datafusion", schema: "information_schema", table: "tables" }), name: "table_schema" }, valid_fields: [Column { relation: Some(Partial { schema: "information_schema", table: "tables" }), name: "table_catalog" }, Column { relation: Some(Partial { schema: "information_schema", table: "tables" }), name: "table_schema" }, Column { relation: Some(Partial { schema: "information_schema", table: "tables" }), name: "table_name" }, Column { relation: Some(Partial { schema: "information_schema", table: "tables" }), name: "table_type" }] }, Some(""))
```

So DF asks fully qualified column name in the list of partially qualified ones., imho this test shouldn't pass. 
We typically allow less qualified to find a match for more qualified like `a -> t1.a` or `a -> schema.t1.a` but not the opposite.

I'm open to discussions.

Reply:
I think it should still work.

If the default catalog is `datafusion`, then all tables that don't have a catalog specified should implicitly fall under `datafusion` catalog. So querying `information_schema.tables` is the same as querying `datafusion.information_schema.tables` which means we should be able to use `datafusion.information_schema.tables.table_schema` in the where clause and resolve it.

I think I've worked on or seen an issue related to this before, and I agree it might be a bit confusing.

Reply:
Oh you mean if the catalog is not specified, by default we should consider `catalog.default_catalog` value? 
Would you mind if I addresss it in followup PR, I wanna keep this PR size to be reviewable

Reply:
Yeah that sounds good :+1: